### ISSUE TITLE
Using swa fails in imdb_scripts

### ISSUE THREAD

**Issue by samuelharden** [fastai/fastai#854]
**Describe the bug**
While using use_swa feature of a RNNLearner, i am getting exception about copying.

Code in question:
```
        if use_swa:
            # make a copy of the model to track average weights
            self.swa_model = copy.deepcopy(model)

```

Error while running:
```
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/scratch/gaurava/miniconda/envs/fastai/lib/python3.6/copy.py", line 161, in deepcopy
    y = copier(memo)
  File "/scratch/gaurava/miniconda/envs/fastai/lib/python3.6/site-packages/torch/autograd/variable.py", line 91, in __deepcopy__
    raise RuntimeError("Only Variables created explicitly by the user "
RuntimeError: Only Variables created explicitly by the user (graph leaves) support the deepcopy protocol at the moment

```
**To Reproduce**
Enable `use_swa` in RNNLearned in `imdb_scripts` of fastai

```
-    learn.fit(lrs, n_cycles, wds=wd, cycle_len=cl, use_clr=(8,8) if use_clr else None)
+    learn.fit(lrs, n_cycles, wds=wd, cycle_len=cl, use_clr=(8,8) if use_clr else None, use_swa=use_swa)

```
**Expected behavior**
Should work as expected and not throw an error.

**Comment by samuelharden** [comment]
Since it's a deep copy I realized might be worth copying whole stacktrace: https://pastebin.com/3yW7mi1A

Moreover, it seems to appear from here https://github.com/pytorch/pytorch/blob/5760b036fb338eacd641418321f23aee51b1aee9/torch/autograd/variable.py#L89-L97

**Comment by sgugger** [comment]
Thanks for all the details. I'm afraid this won't be fixed in 0.7 as we've moved on to v1. You already have the language models there (a bit of rewrite is necessary since v1 isn't backward compatible) and SWA will soon be implemented (I think it's also a pytorch optimizer now, if that helps).
